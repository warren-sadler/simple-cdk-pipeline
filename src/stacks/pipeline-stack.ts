import { Artifact, Pipeline } from "@aws-cdk/aws-codepipeline";
import { GitHubSourceAction } from "@aws-cdk/aws-codepipeline-actions";
import { SecretValue } from "@aws-cdk/core";
import * as cdk from "@aws-cdk/core";

export class PipelineStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);
    const pipeline = new Pipeline(this, "SimplePipeline", {
      pipelineName: "SimplePipeline",
      crossAccountKeys: false,
    });
    const sourceOutput = new Artifact("SourceOutput");
    pipeline.addStage({
      stageName: "Source",
      actions: [
        new GitHubSourceAction({
          owner: "warren-sadler",
          repo: "simple-cdk-pipeline",
          branch: "main",
          actionName: "Pipeline Source",
          oauthToken: SecretValue.secretsManager(
            "simple-cdk-pipeline-gh-token"
          ),
          output: sourceOutput,
        }),
      ],
    });
  }
}
