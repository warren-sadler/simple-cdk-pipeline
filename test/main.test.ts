import { SynthUtils } from "@aws-cdk/assert";
import { App } from "@aws-cdk/core";
import { BillingStack } from "../src/stacks/billing-stack";

test("Snapshot", () => {
  const app = new App();
  const stack = new BillingStack(app, "test", {
    budgetAmount: 10,
    emailAddress: "test@test.com",
  });
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
