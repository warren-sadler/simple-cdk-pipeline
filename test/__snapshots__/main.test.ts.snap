// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Object {
  "Resources": Object {
    "BudgetED13AE41": Object {
      "Properties": Object {
        "Budget": Object {
          "BudgetLimit": Object {
            "Amount": 10,
            "Unit": "USD",
          },
          "BudgetName": "Monthly Budget",
          "BudgetType": "COST",
          "TimeUnit": "MONTHLY",
        },
        "NotificationsWithSubscribers": Array [
          Object {
            "Notification": Object {
              "ComparisonOperator": "GREATER_THAN",
              "NotificationType": "ACTUAL",
              "Threshold": 100,
              "ThresholdType": "PERCENTAGE",
            },
            "Subscribers": Array [
              Object {
                "Address": "test@test.com",
                "SubscriptionType": "EMAIL",
              },
            ],
          },
        ],
      },
      "Type": "AWS::Budgets::Budget",
    },
  },
}
`;
